<html>
<!--#include virtual="../header.shtml" -->

<head>
  <title>RADIANT Project</title>
</head>
<body>

<!--#include virtual="../navigation.shtml" -->

<section id="content" class="three-col">
<div id="inner"><h1>RADIANT: Rapid Development and Distribution of Statistical Tools for High-Throughput Sequencing Data</h1>
<h2>Overview</h2><p><p>RADIANT is a research project funded by the 7th Framework Programme of the European Commission.The project started in December 2012 and brings together ten project partners from five European countries.

<p>The aim of the RADIANT project is to develop new statistical analysis tools for solving open problems in high-throughput sequencing (HTS) data analysis. HTS is a rapidly evolving family of technologies with many applications, including genetics of both rare and common diseases, understanding disease mechanism progression. RADIANT aims to provide an integrated computational framework for HTS data analysis that is robust and user-friendly, and provide tools to benchmark experimental protocols and statistical methods. The project will establish training materials and an extensive training programme for the rapid dissemination of these new tools to the biomedical community.
<p>The project is sponsored by <a href="http://cordis.europa.eu/projects/rcn/105701_en.html">EU FP7-HEALTH Project Ref 305626 "Rapid development and distribution of statistical tools for high-throughput sequencing data"</a> and is a collaboration with <a href="http://www.ls.manchester.ac.uk/people/profile/?personid=10584">Magnus Rattray</a> of University of Manchester, <a href="http://radiant-project.eu/Consortium/Partners/Genomatix.html">Korbinian Grote</a> of Genomatix Software GmbH Germany, <a href="http://radiant-project.eu/Consortium/Partners/Genomnia.html">Alessandro Guffanti</a> of Genomia Italy, <a href="http://radiant-project.eu/Consortium/Partners/EMBL.html">Wolfgang Huber</a> of EMBL Heidelberg, <a href="http://dibernardo.tigem.it/wiki/index.php/Main_Page">Diego di Bernardo</a> of Fondazione Telethon, <a href="http://radiant-project.eu/Consortium/Partners/IIT.html">Mattia Pelizzola</a> of Istituto Italiano di Tecnologia, <a href="http://radiant-project.eu/Consortium/Partners/ARMINES.html">Jean-Philippe Vert</a> of ARMINES France, <a href="http://www.insilico-biotechnology.com/people">Klaus Mauch</a> of Insilico Biotechnology, <a href="http://www.fluxome.com/company/organization/leadership-team.aspx">Jean-Marie Mouillon</a> of Fluxome Sciences A/S, <a href="http://radiant-project.eu/Consortium/Partners/Zurich.html">Mark Robinson</a> of University of Zurich Switzerland and <a href="http://radiant-project.eu/Consortium/Partners/CAM.html">Simon Tavare</a> of University of Cambridge.<a name="personnel"></a><h2>Personnel at Sheffield</h2>

<table>
<tr><td><a href="http://www.dcs.sheffield.ac.uk/cgi-bin/makeperson?Z.Dai">Zhenwen Dai</a>, post-doctoral research assistant</td></tr>
<tr><td><a href="http://www.dcs.sheffield.ac.uk/cgi-bin/makeperson?T.deCampos">Teo de Campos</a>, post-doctoral research assistant</td></tr>
<tr><td><a href="http://www.dcs.sheffield.ac.uk/cgi-bin/makeperson?J.Hensman">James Hensman</a>, MRC Fellow</td></tr>
</table>

<a name="publications"></a><h2>Publications</h2>

<p>The following publications have provided background to our work in this project.<h3>Conference Papers</h3>

<p><a name="Hensman:tvb14"></a><span class=author>J. Hensman, M. Zwiessele and N. D. Lawrence. </span> (2014) <span class=papertitle>"Tilted variational Bayes"</span> in S. Kaski and J. Corander (eds) <span class=journal>Proceedings of the Seventeenth International Workshop on Artificial	Intelligence and Statistics</span>, JMLR W&CP 33, Iceland, pp . [<a href="https://github.com/SheffieldML/GPy " target="_blank">Software</a>][<a href="http://scholar.google.com/scholar?hl-en&lr=&q=Tilted+Variational+Bayes+&btnG=Search" target="_blank">Google Scholar Search</a>]

<h4>Abstract</h4>

<p class=abstract>We present a novel method for approximate inference. Using some of the constructs from expectation propagation (EP), we derive a lower bound of the marginal likelihood in a similar fashion to variational Bayes (VB). The method combines some of the benefits of VB and EP: it can be used with light-tailed likelihoods (where traditional VB fails), and it provides a lower bound on the marginal likelihood. We apply the method to Gaussian process classification, a situation where the Kullback-Leibler divergence minimized in traditional VB can be infinite, and to robust Gaussian process regression, where the inference process is dramatically simplified in comparison to EP.<br><br>Code to reproduce all the experiments can be found at <a href="github.com/SheffieldML/TVB"><font face="Courier New">github.com/SheffieldML/TVB</font></a>.
</p>



<p><a name="Andrade:hybrid14"></a><span class=author>R. Andrade-Pacheco, J. Hensman and N. D. Lawrence. </span> (2014) <span class=papertitle>"Hybrid discriminative-generative approaches with Gaussian processes"</span> in S. Kaski and J. Corander (eds) <span class=journal>Proceedings of the Seventeenth International Workshop on Artificial	Intelligence and Statistics</span>, JMLR W&CP 33, Iceland, pp . [<a href="https://github.com/SheffieldML/GPy " target="_blank">Software</a>][<a href="http://scholar.google.com/scholar?hl-en&lr=&q=Hybrid+Discriminative-Generative+Approaches+with+Gaussian+Processes+&btnG=Search" target="_blank">Google Scholar Search</a>]

<h4>Abstract</h4>

<p class=abstract>Machine learning practitioners are often faced with a choice between a discriminative and a generative approach to modelling. Here, we present a model based on a hybrid approach that breaks down some of the barriers between the discriminative and generative points of view, allowing continuous dimensionality reduction of hybrid discrete-continous data, discriminative classification with missing inputs and manifold learning informed by class labels.
</p>



<h3>Related References</h3>

<p><a name="Hensman:hierarchical13"></a><span class=author>J. Hensman, N. D. Lawrence and M. Rattray. </span> (2013) <span class=papertitle>"Hierarchical Bayesian modelling of gene expression time series across irregularly sampled replicates and clusters"</span> in <span class=journal>BMC Bioinformatics</span>  14 (252)  [<a href="http://dx.doi.org/doi:10.1186/1471-2105-14-252"target="_blank">DOI</a>][<a href="http://scholar.google.com/scholar?hl-en&lr=&q=Hierarchical+Bayesian+modelling+of+gene+expression+time+series+across+irregularly+sampled+replicates+and+clusters+&btnG=Search" target="_blank">Google Scholar Search</a>]

<h4>Abstract</h4>

<p class=abstract>\textbf{Background}<br><br>Time course data from microarrays and high-throughput sequencing experiments require simple, computationally efficient and powerful statistical models to extract meaningful biological signal, and for tasks such as data fusion and clustering. Existing methodologies fail to capture either the temporal or replicated nature of the experiments, and often impose constraints on the data collection process, such as regularly spaced samples, or similar sampling schema across replications.<br><br>\textbf{Results}<br><br>We propose hierarchical Gaussian processes as a general model of gene expression time-series, with application to a variety of problems. In particular, we illustrate the method's capacity for missing data imputation, data fusion and clustering.The method can impute data which is missing both systematically and at random: in a hold-out test on real data, performance is significantly better than commonly used imputation methods. The method's ability to model inter- and intra-cluster variance leads to more biologically meaningful clusters. The approach removes the necessity for evenly spaced samples, an advantage illustrated on a developmental Drosophila dataset with irregular replications.<br><br>\textbf{Conclusion}<br><br>The hierarchical Gaussian process model provides an excellent statistical basis for several gene-expression time-series tasks. It has only a few additional parameters over a regular GP, has negligible additional complexity, is easily implemented and can be integrated into several existing algorithms. Our experiments were implemented in python, and are available from the authors' website: <a href="http://staffwww.dcs.shef.ac.uk/people/J.Hensman/"><font face="Courier New">http://staffwww.dcs.shef.ac.uk/people/J.Hensman/</font></a>.
</p>



<p><a name="Fusi:detecting13"></a><span class=author>N. Fusi, C. Lippert, K. Borgwardt, N. D. Lawrence and O. Stegle. </span> (2013) <span class=papertitle>"Detecting regulatory gene-environment interactions with unmeasured environmental factors"</span> in <span class=journal>Bioinformatics</span>   [<a href="http://dx.doi.org/10.1093/bioinformatics/btt148"target="_blank">DOI</a>][<a href="http://scholar.google.com/scholar?hl-en&lr=&q=Detecting+Regulatory+Gene-Environment+Interactions+with+Unmeasured+Environmental+Factors+&btnG=Search" target="_blank">Google Scholar Search</a>]

<h4>Abstract</h4>

<p class=abstract>\textbf{Motivation}: Genomic studies have revealed a substantial heritable component of the transcriptional state of the cell. To fully understand the genetic regulation of gene expression variability, it is important to study the effect of genotype in the context of external factors such as alternative environmental conditions. In model systems, explicit environmental perturbations have been considered for this purpose, allowing to directly test for environment-specific genetic effects. However, such experiments are limited to species that can be profiled in controlled environments, hampering their use in important systems such as human. Moreover, even in seemingly tightly regulated experimental conditions, subtle environmental perturbations cannot be ruled out, and hence unknown environmental influences are frequent. Here, we propose a model-based approach to simultaneously infer unmeasured environmental factors from gene expression profiles and use them in genetic analyses, identifying environment-specific associations between polymorphic loci and individual gene expression traits.<br><br> \textbf{Results}: In extensive simulation studies, we show that our method is able to accurately reconstruct environmental factors and their interactions with genotype in a variety of settings. We further illustrate the use of our model in a real-world dataset in which one environmental factor has been explicitly experimentally controlled. Our method is able to accurately reconstruct the true underlying environmental factor even if it's not given as an input, allowing to detect genuine genotype-environment interactions. In addition to the known environmental factor, we find unmeasured factors involved in novel genotype-environment interactions. Our results suggest that interactions with both known and unknown environmental factors significantly contribute to gene expression variability.<br><br> \textbf{Availability}: Software available at <a href="http://ml.sheffield.ac.uk/qtl/limmi"><font face="Courier New">http://ml.sheffield.ac.uk/qtl/limmi</font></a><br><br> \textbf{Contact}: <a href="oliver.stegle@ebi.ac.uk"><font face="Courier New">oliver.stegle@ebi.ac.uk</font></a>, <a href="nicolo.fusi@sheffield.ac.uk"><font face="Courier New">nicolo.fusi@sheffield.ac.uk</font></a>
</p>



<p><a name="Hensman:fast12"></a><span class=author>J. Hensman, M. Rattray and N. D. Lawrence. </span> (2012) <span class=papertitle>"Fast variational inference in the conjugate exponential family"</span> in P. L. Bartlett, F. C. N. Pereira, C. J. C. Burges, L &#233o. Bottou and K. Q. Weinberger (eds) <span class=journal>Advances in Neural Information Processing Systems</span>, . [<a href=http://books.nips.cc/papers/files/nips25/NIPS2012_1314.pdf>PDF</a>][<a href="http://scholar.google.com/scholar?hl-en&lr=&q=Fast+variational+inference+in+the+Conjugate+Exponential+family+&btnG=Search" target="_blank">Google Scholar Search</a>]</p>



<p><a name="Hensman:bigdata13"></a><span class=author>J. Hensman, N. Fusi and N. D. Lawrence. </span> (2013) <span class=papertitle>"Gaussian processes for big data"</span> in A. Nicholson and P. Smyth (eds) <span class=journal>Uncertainty in Artificial Intelligence</span>, AUAI Press, . [<a href=http://auai.org/uai2013/prints/papers/244.pdf>PDF</a>][<a href="http://scholar.google.com/scholar?hl-en&lr=&q=Gaussian+Processes+for+Big+Data+&btnG=Search" target="_blank">Google Scholar Search</a>]</p>



<p><a name="Fusi:genomics12"></a><span class=author>N. Fusi, O. Stegle and N. D. Lawrence. </span> (2012) <span class=papertitle>"Joint modelling of confounding factors and prominent genetic regulators provides increased accuracy in genetical genomics studies"</span> in <span class=journal>PLoS Computat Biol</span>  8, pp e1002330  [<a href="http://ml.sheffield.ac.uk/qtl/ " target="_blank">Software</a>][<a href=http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1002330>PDF</a>][<a href="http://dx.doi.org/10.1371/journal.pcbi.1002330"target="_blank">DOI</a>][<a href="http://scholar.google.com/scholar?hl-en&lr=&q=Joint+Modelling+of+Confounding+Factors+and+Prominent+Genetic+Regulators+Provides+Increased+Accuracy+in+Genetical+Genomics+Studies+&btnG=Search" target="_blank">Google Scholar Search</a>]

<h4>Abstract</h4>

<p class=abstract>Expression quantitative trait loci (eQTL) studies are an integral tool to investigate the genetic component of gene expression variation. A major challenge in the analysis of such studies are hidden confounding factors, such as unobserved covariates or unknown subtle environmental perturbations. These factors can induce a pronounced artifactual correlation structure in the expression profiles, which may create spurious false associations or mask real genetic association signals. Here, we report PANAMA (Probabilistic ANAlysis of genoMic dAta), a novel probabilistic model to account for confounding factors within an eQTL analysis. In contrast to previous methods, PANAMA learns hidden factors jointly with the effect of prominent genetic regulators. As a result, this new model can more accurately distinguish true genetic association signals from confounding variation. We applied our model and compared it to existing methods on different datasets and biological systems. PANAMA consistently performs better than alternative methods, and finds in particular substantially more trans regulators. Importantly, our approach not only identifies a greater number of associations, but also yields hits that are biologically more plausible and can be better reproduced between independent studies. A software implementation of PANAMA is freely available online at <a href="http://ml.sheffield.ac.uk/qtl/"><font face="Courier New">http://ml.sheffield.ac.uk/qtl/</font></a>.
</p>



<p>This document last modified <!--#flastmod file="index.html" --></p>
</div>
</section><!--#include virtual="../footer.shtml" -->


</html>